drivers/storage/portworx/portworx.go

```diff
@@ -712,14 +998,27 @@ func SetPortworxDefaults(toUpdate *corev1.StorageCluster, k8sVersion *version.Ve
 			toUpdate.Spec.Storage.UseAll = boolPtr(true)
 		}
 	}
+}
 
-	if pxutil.IsTelemetryEnabled(toUpdate.Spec) && t.pxVersion.LessThan(pxutil.MinimumPxVersionCCM) {
-		toUpdate.Spec.Monitoring.Telemetry.Enabled = false // telemetry not supported for < 2.8
-		toUpdate.Spec.Monitoring.Telemetry.Image = ""
+func setPortworxCloudStorageSpecDefaults(toUpdate *corev1.StorageCluster) {
+	if preflight.IsEKS() {
+		if toUpdate.Spec.CloudStorage == nil {
+			toUpdate.Spec.CloudStorage = &corev1.CloudStorageSpec{}
+		}
+		if toUpdate.Spec.CloudStorage.DeviceSpecs == nil {
+			defaultEKSCloudStorageDevice := "type=gp3,size=150"
+			deviceSpecs := []string{defaultEKSCloudStorageDevice}
+			toUpdate.Spec.CloudStorage.DeviceSpecs = &deviceSpecs
+			if toUpdate.Spec.Kvdb.Internal && toUpdate.Spec.CloudStorage.KvdbDeviceSpec == nil {
+				toUpdate.Spec.CloudStorage.KvdbDeviceSpec = &defaultEKSCloudStorageDevice
+			}
```

Comment:
Can we skip this? Portworx already creates a kvdb drive even if we don't specify it.